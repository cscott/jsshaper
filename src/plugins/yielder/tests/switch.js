var log = (typeof console !== "undefined") && console.log || print;
(typeof require !== 'undefined') && require('setup.js');

function l(v) {
  log(v);
  return v;
}

function foo(v) {
  switch (l(v)) {
  case l(1): yield -1;
  case l(2): yield -2;
             break;
  case l(3): { yield -3; }
  default:   yield -4;
    if (l(v)==4) break;
    yield -5;
  case l(6): yield -6;
  }
  log('bottom');
}

function bar(v) {
  var arr = [];
  for (var p in foo(v))
    arr.push(''+p);
  log(arr.join(','));
}

bar(1);
bar(2);
bar(3);
bar(4);
bar(5);
bar(6);
