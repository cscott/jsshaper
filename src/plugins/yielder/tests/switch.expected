1
1
bottom
-1,-2
2
1
2
bottom
-2
3
1
2
3
3
bottom
-3,-4,-5,-6
4
1
2
3
6
4
bottom
-4
5
1
2
3
6
5
bottom
-4,-5,-6
6
1
2
3
6
bottom
-6
